src/test/scala/io/iohk/ethereum/proof/MptProofVerifier.scala

```diff
@@ -0,0 +1,52 @@
+package io.iohk.ethereum.proof
+
+import akka.util.ByteString
+import cats.syntax.either._
+import cats.syntax.functor._
+import io.iohk.ethereum.db.dataSource.EphemDataSource
+import io.iohk.ethereum.db.storage.{NodeStorage, SerializingMptStorage, StateStorage}
+import io.iohk.ethereum.jsonrpc.{MptProofError, KeyNotFoundInRebuidMpt, UnableRebuildMpt}
+import io.iohk.ethereum.mpt.{ByteArrayEncoder, ByteArraySerializable, MerklePatriciaTrie, MptNode}
+
+object MptProofVerifier {
+
+  def verifyProof[K, V](
+      rootHash: Array[Byte],
+      key: K,
+      proof: Vector[MptNode]
+  )(implicit kSer: ByteArrayEncoder[K], vSer: ByteArraySerializable[V]): Either[MptProofError, Unit] = {
```

Comment:
why Unit was chosen as a return type for a verified proof?

Reply:
Now it is ADT.

Reply:
Thank for this remark 👍